test/README.md

```diff
@@ -0,0 +1,65 @@
+# Test
+
+This directory contains e2e tests and testing docs:
+
+- Unit tests are in the code base alongside the code they test
+- e2e tests are in [`test/e2e/`](./e2e)
+
+## Running unit tests
+
+To run all unit tests:
+
+```bash
+$ hack/build.sh -t
+```
+
+_By default `go test` will not run [the e2e tests](#running-e2e-tests-locally),
+which need [`-tags=e2e`](#running-end-to-end-tests) to be enabled._
+
+## Running e2e tests locally
+
+To run [the e2e tests](./e2e) , you need to have a 
+
+1. [Running knative environment.](./../DEVELOPMENT.md#create-a-cluster)
+2. `kn` binary in the $PATH.
+
+Before running the e2e tests please make sure you dont have any namespaces with the name starting with `kne2etests`
```

Comment:
Might want to mention that their `KUBECONFIG` env variable needs to be set or that `~/.kube/config` file is pointing to the cluster's configuration.

Reply:
Added